### DISCUSSION TITLE
Changing tab focus

### DISCUSSION THREAD

**Discussion by carlagomesah** [rubycdp/ferrum#358] [Q&A]
Hi there.
I have this example snippet that works with 2 tabs. Indeed, it correctly recognises each tab but the visual focus doesn't change to `page_one` when going to the last url - `page_one.go_to("page_one.go_to("https://github.com/rubycdp/ferrum")`.  Ideally, instead of going to another url it would click on button on the current url in page-one -  `page_one.at_xpath("/html/body/div[1]/div[6]/div/main/turbo-frame/div/div/div/div[2]/div[1]/div[1]/a").click` - but when I try it fails, I guess because the tab doesn't have focus.

What am I missing?

```
  browser = Ferrum::Browser.new(timeout: 90, headless: false, browser_options: { "disable-blink-features": "AutomationControlled" })
  page_one = browser.create_page
  page_one.go_to("https://www.google.pt")
  page_two = browser.create_page
  page_two.go_to("https://www.google.com")
  page_one.go_to("https://github.com/rubycdp/ferrum")

  # page_one.at_xpath("/html/body/div[1]/div[6]/div/main/turbo-frame/div/div/div/div[2]/div[1]/div[1]/a").click
  sleep(20)
```

I also tried to change active tab with keyboard like:

  `browser.keyboard.down([:control, :tab])` or `page_two.keyboard.down([:control, :tab])` but nothing happens

**Comment by route** [comment]
It's not working because you are using wrong xpath. If you check xpath the last div[1] would be `js-socket-channel js-updatable-content` which is inserted by JS, and it doesn't have any links resulting in `at_xpath` returning `nil`.
